unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWCreatorController.cs

```diff
@@ -208,23 +211,23 @@ private void AddSmartItemComponent(DCLBuilderInWorldEntity entity)
         string jsonModel = JsonUtility.ToJson(model);
 
         //Note (Adrian): This shouldn't work this way, we should have a function to create the component from Model directly
-        sceneToEdit.EntityComponentCreateOrUpdateFromUnity(entity.rootEntity.entityId, CLASS_ID_COMPONENT.SMART_ITEM, jsonModel);
+        sceneToEdit.EntityComponentCreateOrUpdateWithModel(entity.rootEntity.entityId, CLASS_ID_COMPONENT.SMART_ITEM, jsonModel);
 
```

Comment:
If there is a new entitycomponentcreate with model we should use the `model `instead of the `jsonModel`

Reply:
Also, we can remove the comment if it is the case!